Cookbook/Technical-Documents/AutomationIdentifiers.md

```diff
@@ -67,23 +67,90 @@ fileprivate enum NavigationBar {
 
 While this will fix the test, it is susceptible to future changes. In this case time permitting, we should attempt to define an `accessibilityIdentifier` for the element and then update the enum to use this identifier. This is not always possible as in the case with chat responses, but it is something we should strive for in order to make our tests more stable and maintainable, reducing the amount of updates needed in the future.
 
-#### Updating Queries
+### Using AccessibilityContent
+
+It is now possible to directly use accessibility identifiers defined during implementation time without resorting to raw strings.
+
+There is a namespace `AccessibilityContent` in `BabylonDependencies` that should be used by developers to define accessibility identifiers for controls, grouped by screen, for example for Promo Code Entry screen we define a `PromoCode` enum with child controls:
+
+``` swift
+public enum AccessibilityContent {
+    public enum PromoCode: String {
+        case codeCell
+    }
+}
+```
+
+After this identifier has been set on a control, we can `import BabylonDependencies` in a Screen implementation in automation, and update the query to use this identifier:
+
+``` swift
+let promofield = app.tables
+    .cells[AccessibilityContent.PromoCode.codeCell]
+    .textFields
+    .firstMatch
+```
+
+Note that the identifier is usually set on the cell itself rather than its internal components, as automation has problems finding elements otherwise.
+
+### Taking advantage of ScreenNaming for screen identifiers
```

Comment:
I find this section very confusing and I'd prefer if it would only mention that Bento is doing it this way and how then it should be reflected in the automation. And a note that ScreenNaming should not be used for setting accessibility identifiers manually, instead it should use AccessibilityContent.

P.S. IMO Bento should not assign accessibility identifiers automatically especially using something from completely different domain (analytics) and instead offer a way to set accessibility identifier for the screen manually. But that's not a discussion about Bento.

Reply:
Yes, probably makes sense to downplay it a little bit.

Reply:
Following discussion on Slack, mention avoiding using `ScreenNaming` raw identifiers without setting them manually as accessibility identifiers, as automatic setting of them will likely be removed.